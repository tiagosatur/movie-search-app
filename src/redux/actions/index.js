export * from './movies';
export * from './details';