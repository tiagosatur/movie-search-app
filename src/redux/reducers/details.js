import { 
    MOVIE_DETAILS_PENDING,
    MOVIE_DETAILS_SUCCESS,
    MOVIE_DETAILS_ERROR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    details: {},
    error: null
}

export default ( state = initialState, action ) => {
    const { type, details, error } = action;

    switch(type) {
        case MOVIE_DETAILS_PENDING:
            return {
                ...state,
                details: {},
                pending: true
            };
        case MOVIE_DETAILS_SUCCESS:
            return {
                ...state,
                pending: false,
                details: details,
            };
        case MOVIE_DETAILS_ERROR:
            return {
                ...state,
                details: {},
                pending: false,
                error: error
            };
        default: 
            return state;
    }
    
};