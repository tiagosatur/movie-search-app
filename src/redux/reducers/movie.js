import { 
    SEARCH_MOVIE_PENDING,
    SEARCH_MOVIE_SUCCESS,
    SEARCH_MOVIE_ERROR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    list: [],
    totalResults: 0,
    error: null
}

export const movie = ( state = initialState, action ) => {
    const { type, list, error } = action;

    switch(type) {
        case SEARCH_MOVIE_PENDING:
            return {
                ...state,
                list: [],
                totalResults: 0,
                pending: true
            };
        case SEARCH_MOVIE_SUCCESS:
            return {
                ...state,
                pending: false,
                list: list.Search,
                totalResults: list.totalResults,
                error: null,
            };
        case SEARCH_MOVIE_ERROR:
            return {
                ...state,
                list: [],
                pending: false,
                totalResults: 0,
                error: error
            };
        default: 
            return state;
    }
};