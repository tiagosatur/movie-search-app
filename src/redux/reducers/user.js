import { 
    LOGIN_USER_PENDING,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_ERROR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    isLogged: false,
    user: {},
    error: null
}

export const user = ( state = initialState, action ) => {
    const { type, user, error } = action;

    switch(type) {
        case LOGIN_USER_PENDING:
            return {
                ...state,
                user: {},
                pending: true
            };
        case LOGIN_USER_SUCCESS:
            return {
                ...state,
                pending: false,
                isLogged: true,
                user: user,
            };
        case LOGIN_USER_ERROR:
            return {
                ...state,
                pending: false,
                isLogged: false,
                user: {},
                error: error
            };
        default: 
            return state;
    }
};