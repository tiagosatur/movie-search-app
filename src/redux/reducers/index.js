import { combineReducers } from 'redux'
import {movie} from './movie'
import {details} from './details'
import {user} from './user'

export default combineReducers ({
    movie,
    details,
    user,
})