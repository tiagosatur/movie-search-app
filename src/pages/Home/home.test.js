import React from 'react';
import {  Provider } from 'react-redux'
import { initialState } from '../../redux/reducers/movie';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Home from './Home';
import {  MemoryRouter, StaticRouter } from "react-router";

import { createStore } from 'redux';
import rootReducer from '../../redux/reducers'

describe('Home Page tests', () => {
    let store = createStore(rootReducer, initialState);
    
    const wrapper =  mount(
        <Provider store={store}>
            <StaticRouter>
                <Home />
            </StaticRouter>
        </Provider>
    )

    it('should call Home snapshot correctly', () => {
        const tree = renderer
            .create(wrapper)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
