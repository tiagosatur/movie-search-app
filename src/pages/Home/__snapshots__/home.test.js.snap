// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page tests should call Home snapshot correctly 1`] = `
<div>
  <h2>
    Home
  </h2>
  <p>
    Wecome to the application main page
  </p>
</div>
`;
